pywinauto/macos/application.py

```diff
@@ -0,0 +1,239 @@
+import os
+import time
+import sys
+from os import path
+import subprocess
+from subprocess import Popen, PIPE
+
+from AppKit import NSWorkspaceLaunchNewInstance, NSWorkspaceLaunchAllowingClassicStartup
+from Foundation import NSAppleEventDescriptor
+from ApplicationServices import AXUIElementCreateApplication
+
+from . import macos_functions
+from . import ax_element_info
+from .. import backend
+from ..backend import registry
+from ..element_info import ElementInfo
+from ..base_wrapper import BaseWrapper
+from ..base_application import AppStartError, ProcessNotFoundError, AppNotConnected, BaseApplication
+
+
+sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
+parent_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(parent_dir)
+os.path.join
+from ..timings import Timings, wait_until
+backend.register('ax', ElementInfo, BaseWrapper)
+
+
+class Pidmonitor():
+    
+    pid = None
+    pid_before = []
+
+    @staticmethod
+    def pid_before():
+        liste1 = []
+        for apps in macos_functions.running_applications():
+            liste1.append(apps.processIdentifier()) 
+        return liste1
+
+    @staticmethod        
+    def pid_after():
+        liste2 = [] 
+        ax_element_info.cache_update()
+        for apps in macos_functions.running_applications():
+            liste2.append(apps.processIdentifier())
+        return liste2
+
+
+class Application(BaseApplication):
+
+    Pidmonitor.pid_before = Pidmonitor.pid_before()
+
+    def __init__(self, backend="ax"):
+        self.connected = False
+        self.ns_app = None
+        self.process = None
+        if backend not in registry.backends:
+            raise ValueError('Backend "{0}" is not registered!'.format(backend))
+        self.backend = registry.backends[backend]
+
+    def start(self, name=None, bundle_id=None, new_instance=True):
+
+        if name is not None and bundle_id is not None:
+              raise ValueError('Parameters name and bundle_id are mutually exclusive. Use only one of them at the moment.')
+
+        if name is not None:
+            bundle = macos_functions.bundle_identifier_for_application_name(name)
+            print(bundle)
+            macos_functions.launch_application_by_bundle(bundle, new_instance)
+            NsArray = macos_functions.get_app_instance_by_bundle(bundle)
+
+            self.ns_app = NsArray[0]
```

Comment:
`NsArray` name looks like a type and it might be simply 
`self.n_app = macos_functions.get_app_instance_by_bundle(bundle)`
Similar is down on line 92 too